pkg/cloudprovider/aws/factory.go

```diff
@@ -25,39 +25,36 @@ import (
 	"github.com/aws/aws-sdk-go/service/eks"
 	"github.com/aws/aws-sdk-go/service/eks/eksiface"
 	"github.com/aws/aws-sdk-go/service/iam"
-	"github.com/aws/aws-sdk-go/service/iam/iamiface"
 	"github.com/aws/aws-sdk-go/service/sqs"
 	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
 	"github.com/awslabs/karpenter/pkg/apis/autoscaling/v1alpha1"
+	provisioningv1alpha1 "github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
 	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/aws/fleet"
 	"github.com/awslabs/karpenter/pkg/cloudprovider/fake"
 	"github.com/awslabs/karpenter/pkg/utils/log"
-	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Factory struct {
 	AutoscalingClient autoscalingiface.AutoScalingAPI
-	SQS               sqsiface.SQSAPI
-	EKS               eksiface.EKSAPI
-	EC2               ec2iface.EC2API
-	IAM               iamiface.IAMAPI
-	Client            client.Client
-	// TODO, dedup this with client. Currently necessary due to the client not
-	// working until mgr.Start() is called, which breaks initialization logic.
-	Config *rest.Config
+	sqs               sqsiface.SQSAPI
+	eks               eksiface.EKSAPI
+	ec2               ec2iface.EC2API
+	fleetFactory      *fleet.Factory
+	kubeClient        client.Client
 }
 
 func NewFactory(options cloudprovider.Options) *Factory {
 	sess := withRegion(session.Must(session.NewSession()))
+	EC2 := ec2.New(sess)
 	return &Factory{
 		AutoscalingClient: autoscaling.New(sess),
-		EKS:               eks.New(sess),
-		SQS:               sqs.New(sess),
-		EC2:               ec2.New(sess),
-		IAM:               iam.New(sess),
-		Client:            options.Client,
-		Config:            options.Config,
+		eks:               eks.New(sess),
+		sqs:               sqs.New(sess),
+		ec2:               EC2,
```

Comment:
To maintain the consistency with other APIs, can we call `ec2.New(sess)` here.

Reply:
I'm reusing it from line 50. I want a single ec2 client for all of karpenter, but it's required for the node group controller, as well as the fleet factory. That's why it's separated and injected in both places.